client/src/main/java/org/eclipse/hono/client/impl/HonoClientImpl.java

```diff
@@ -642,7 +658,17 @@ private void removeCredentialsClient(final String tenantId) {
         return result;
     }
 
-    private Future<RequestResponseClient> newRegistrationClient(final String tenantId) {
+    /**
+     * Custom implementation of {@link RegistrationClient} can be instantiated by overriding this
+     * method. Any such instance should be scoped to the given tenantId.
+     *
```

Comment:
We always put a `<p>` after the introduction to separate it from the detailed explanation.